static/js/AddToSourceSheet.jsx

```diff
@@ -89,7 +90,32 @@ class AddToSourceSheetBox extends Component {
       }, this.confirmAdd);
     }
   }
-  addToSourceSheet() {
+
+  longestSuffixPrefixIndex(string1, string2) {
+    let longestSuffixIndex = 0;
+    for (let i = 0; i < string1.length; i++){
+      let suffix = string1.slice(i);
+      if (string2.startsWith(suffix)) {
+        longestSuffixIndex = i;
+      }
+    }
+    return longestSuffixIndex;
+  }
+  longestPrefixSuffixIndex(string1, string2) {
+    let longestPrefixIndex = 0;
+    for (let i = 0; i < string1.length; i++) {
+      let prefix = string1.slice(0, i + 1);
+      if (string2.endsWith(prefix)) {
+        longestPrefixIndex = i + 1;
+      }
+    }
+    return longestPrefixIndex;
+  }
+
+  normalize(text){
+    return(text.replaceAll(/(<br\/>)+/g, ' ').replace(/\u2009/g, ' ').replace(/<[^>]*>/g, '').replace(/\u2009/g, ' '));
```

Comment:
why is `\u2009` repeated twice here? I would instead use `\s` which includes `\u2009` and all other whitespace. This includes newline. Is that ok? see this article for details: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_expressions/Character_classes